include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +512,79 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class ForwardIterator>
+auto sorted_mode(const ForwardIterator first, const ForwardIterator last) -> std::vector<typename std::iterator_traits<ForwardIterator>::value_type>
```

Comment:
If you really want it to be generic, don't return a `std::vector`. Do as all the standard algorithms do: take an output iterator.

Reply:
@jeremy-murphy : Good suggestion, but looking it up I only see it in [C++20](https://en.cppreference.com/w/cpp/iterator/output_iterator).

Reply:
@NAThompson That is pretty idiomatic STL. Looking at the test_num benchmark it is nearly equivalent to returning the vector in terms of complexity. The one caveat is I can not use `benchmark::DoNotOptimize` because sorted_modes would no longer have a return value.

Reply:
@mborland , looks closely at the STL algorithms that take output iterators, they _do_ have a return value.

Reply:
@jeremy-murphy Found them. It had been a bit since I cracked open the C++ Programming Language.